shared/src/test/scala/scorex/crypto/hash/HashTest.scala

```diff
@@ -58,16 +58,16 @@ trait HashTest extends AnyPropSpec
 
     // TODO refactor: reimplement Digest classes to pass runtime check or remove this test
     // test incorrect in Scala2 last check not reached
-//    property(s"${hash.getClass.getSimpleName} should return correct Tag") {
-//      forAll { (string: String, bytes: Array[Byte]) =>
-//        val digest = hash(string)
-//        digest.isInstanceOf[D] shouldBe true
-//        if (digest.isInstanceOf[Digest32]) {
-//          hash.DigestSize shouldBe 32
-//        } else if (digest.isInstanceOf[Digest64]) {
-//          hash.DigestSize shouldBe 64
-//        }
-//      }
-//    }
+    property(s"${hash.getClass.getSimpleName} should return correct Tag") {
+      forAll { (string: String, bytes: Array[Byte]) =>
+        val digest = hash(string)
+        digest.isInstanceOf[D] shouldBe true
+        if (digest.isInstanceOf[Digest32]) {
+          hash.DigestSize shouldBe 32
+        } else if (digest.isInstanceOf[Digest64]) {
+          hash.DigestSize shouldBe 64
+        }
```

Comment:
It looks that again the tests pass, because the code is not reached.

Just add an `else { fail("Tag is not Digest32 or Digest64") 

as demonstrated here:

https://github.com/stenolog/scrypto/pull/3